extensions/oplogPopulator/allocationStrategy/RetainBucketsDecorator.js

```diff
@@ -0,0 +1,86 @@
+const AllocationStrategy = require('./AllocationStrategy');
+
+/**
+ * @class RetainBucketsDecorator
+ *
+ * @classdesc RetainBucketsDecorator is a decorator
+ * that retains buckets that are removed from the
+ * connector but still in use.
+ */
+class RetainBucketsDecorator extends AllocationStrategy {
+    /**
+     * @constructor
+     * @param {AllocationStrategy} strategy the strategy to decorate
+     * @param {Object} params params
+     * @param {Logger} params.logger logger object
+     */
+    constructor(strategy, params) {
+        super(params);
+        this._strategy = strategy;
+
+        // Stores buckets that should are removed from the connector
+        // but still in use
+        this._retainedBuckets = new Map();
+    }
+
+    /**
+     * Callback when a bucket is removed from a connector
+     * @param {String} bucket bucket name
+     * @returns {undefined}
+     */
+    onBucketRemoved(bucket) {
+        // When a bucket is removed, it is not immediately removed
+        // from the connector. This allows to retain the bucket
+        // in the connector until it is removed.
+        this._retainedBuckets.delete(bucket);
```

Comment:
when bucket is removed from the connector, it actually _becomes_ retained

```suggestion
        this._retainedBuckets.set(bucket, connector);
```